pkg/collector/corechecks/cluster/kubernetes_eventbundle.go

```diff
@@ -0,0 +1,84 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+// +build kubeapiserver
+
+package cluster
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/ericchiang/k8s/api/v1"
+)
+
+type kubernetesEventBundle struct {
+	objUid        string
+	readableKey   string
+	component     string
+	events        []*v1.Event
+	timeStamp     float64
+	lastTimestamp float64
+	countByAction map[string]int
+}
+
+func newKubernetesEventBundler(objUid string, compName string) *kubernetesEventBundle {
+	return &kubernetesEventBundle{
+		objUid:        objUid,
+		component:     compName,
+		countByAction: make(map[string]int),
+	}
+}
+
+func (k *kubernetesEventBundle) addEvent(event *v1.Event) error {
+	if *event.InvolvedObject.Uid != k.objUid {
+		return fmt.Errorf("mismatching Object UIDs: %s != %s", event.InvolvedObject.Uid, k.objUid)
+	}
+	k.events = append(k.events, event)
+	k.timeStamp = math.Max(k.timeStamp, float64(*event.Metadata.CreationTimestamp.Seconds))
+	k.lastTimestamp = math.Max(k.timeStamp, float64(*event.LastTimestamp.Seconds))
+
+	if event.Reason == nil || event.Message == nil || event.InvolvedObject.Name == nil || event.InvolvedObject.Kind == nil {
+		return errors.New("could not retrieve some attributes of the event")
+	}
+	k.countByAction[fmt.Sprintf("**%s**: %s\n", *event.Reason, *event.Message)] += int(*event.Count)
```

Comment:
IIRC from my testing, `event.Count` holds the total number of repetitions from the creation of the event.

That means if we got repetitions 4 to 8 in a bucket, we'll have 5 events with counts [4, 5, 6, 7, 8]. I fear `k.countByAction` will hold 30 instead of 5. We should test that

Reply:
If we have a bucket of N events with attributes generating the same aggregation key, the counts will indeed be added and it'll result in Datadog_Event: (Sum(event.count[i], i from 4 to 8) event.Reason event.Message) and this is the desired behaviour in my opinion.

This is what it end up looking like:
https://cl.ly/2D3Q342o0N2v

As we aggregate per ObjUid we care about the underlying counts being aggregated not the number of occurrence of an event during 15 seconds.